sdk/include/asm/trapamd64.inc

```diff
@@ -98,47 +95,32 @@ MACRO(EnterTrap, Flags)
     sub rsp, (KTRAP_FRAME_LENGTH - SIZE_INITIAL_FRAME)
     .allocstack (KTRAP_FRAME_LENGTH - SIZE_INITIAL_FRAME)
 
-    /* Save rbp and rax */
+    /* Save rbp */
     mov [rsp + KTRAP_FRAME_Rbp], rbp
     .savereg rbp, KTRAP_FRAME_Rbp
-    mov [rsp + KTRAP_FRAME_Rax], rax
-    .savereg rax, KTRAP_FRAME_Rax
 
     /* Point rbp to the KTRAP_FRAME */
     lea rbp, [rsp]
     .setframe rbp, 0
 
-    if (Flags AND TF_NONVOLATILES)
-        /* Save non-volatile registers */
-        mov [rbp + KTRAP_FRAME_Rbx], rbx
-        .savereg rbx, KTRAP_FRAME_Rbx
-        mov [rbp + KTRAP_FRAME_Rdi], rdi
-        .savereg rdi, KTRAP_FRAME_Rdi
-        mov [rbp + KTRAP_FRAME_Rsi], rsi
-        .savereg rsi, KTRAP_FRAME_Rsi
-    endif
-
     .endprolog
 
-    if (Flags AND TF_VOLATILES)
-        /* Save volatile registers */
-        mov [rbp + KTRAP_FRAME_Rcx], rcx
-        mov [rbp + KTRAP_FRAME_Rdx], rdx
-        mov [rbp + KTRAP_FRAME_R8], r8
-        mov [rbp + KTRAP_FRAME_R9], r9
-        mov [rbp + KTRAP_FRAME_R10], r10
-        mov [rbp + KTRAP_FRAME_R11], r11
-    endif
-
-    if (Flags AND TF_XMM)
-        /* Save xmm registers */
-        movdqa [rbp + KTRAP_FRAME_Xmm0], xmm0
-        movdqa [rbp + KTRAP_FRAME_Xmm1], xmm1
-        movdqa [rbp + KTRAP_FRAME_Xmm2], xmm2
-        movdqa [rbp + KTRAP_FRAME_Xmm3], xmm3
-        movdqa [rbp + KTRAP_FRAME_Xmm4], xmm4
-        movdqa [rbp + KTRAP_FRAME_Xmm5], xmm5
-    endif
+    /* Save volatile registers */
+    mov [rbp + KTRAP_FRAME_Rax], rax
+    mov [rbp + KTRAP_FRAME_Rcx], rcx
+    mov [rbp + KTRAP_FRAME_Rdx], rdx
+    mov [rbp + KTRAP_FRAME_R8], r8
+    mov [rbp + KTRAP_FRAME_R9], r9
+    mov [rbp + KTRAP_FRAME_R10], r10
+    mov [rbp + KTRAP_FRAME_R11], r11
+
+    /* Save volatile xmm registers */
+    movaps [rbp + KTRAP_FRAME_Xmm0], xmm0
```

Comment:
For what it's worth:
https://stackoverflow.com/questions/6678073/difference-between-movdqa-and-movaps-x86-instructions
```
movaps has smaller machine code (3 bytes minimum): movdqa needs an extra prefix so it's at lest 4 bytes.

On some (but not all) micro-architectures, there are timing differences due to "domain crossing penalties". For this reason, one should generally use movdqa when the data is being used with integer SSE instructions, and movaps when the data is being used with floating-point instructions.

MOVAPS is SSE1 while MOVDQA is SSE2.
```

--

For a potential follow-up, at least trap.S and setjmp.s seem to have similar `movdqa` code.
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=movdqa